CoreGraphNode/src/au/gov/asd/tac/constellation/graph/node/plugins/DefaultPluginEnvironment.java

```diff
@@ -59,67 +58,52 @@
     @Override
     public Future<?> executePluginLater(final Graph graph, final Plugin plugin, final PluginParameters parameters, final boolean interactive, final List<Future<?>> async, final PluginSynchronizer synchronizer) {
 
-        return pluginExecutor.submit(new Callable<Object>() {
-            @Override
-            public Object call() {
-                Thread.currentThread().setName(THREAD_POOL_NAME);
-
-                // If a Future has been specified, don't do anything until the Future has completed.
-                // A typical use-case is an arrangement followed by a camera reset: obviously doing the reset before the
-                // vertices have been relocated is not sensible.
-                if (async != null) {
-                    for (Future<?> future : async) {
-                        if (future != null) {
-                            try {
-                                future.get();
-                            } catch (InterruptedException ex) {
-                                Exceptions.printStackTrace(ex);
-                                Thread.currentThread().interrupt();
-                            } catch (ExecutionException ex) {
-                                Exceptions.printStackTrace(ex);
-                            }
+        return pluginExecutor.submit(() -> {
+            Thread.currentThread().setName(THREAD_POOL_NAME);
```

Comment:
I wouldn't suggest doing it now - but this lambda is big enough to be broken up into several functions.